Reorganize README (#6)
coderaiser/supertape: README.md

[**Tape**](https://github.com/substack/tape)-inspired [`TAP`](https://testanything.org/)-compatible simplest high speed test runner with superpowers.

----

> 

Reviewer: It broken syntax highlighting in VIM

> 

Reviewer: ```suggestion
```

📼 **Supertape** is a fast, minimal test runner with the soul of **tape**. It's designed to be as compatible as possible with **tape** while still having some key improvements, such as:

> - out-of-the-box [ES6 module](https://nodejs.org/api/esm.html) support (take a look at [mock-import](https://github.com/coderaiser/mock-import) for mocking and 🎩[ESCover](https://github.com/coderaiser/escover) for coverage)

Reviewer: ```suggestion
- out-of-the-box [EcmaScript Modules](https://nodejs.org/api/esm.html) support (take a look at [mock-import](https://github.com/coderaiser/mock-import) for mocking and 🎩[ESCover](https://github.com/coderaiser/escover) for coverage)
```

📼 **Supertape** doesn't contain:

- assertion aliases, making the available operators far more concise
- `ES3 code` and lots of [ponyfills](https://github.com/sindresorhus/ponyfill#how-are-ponyfills-better-than-polyfills)
- `t.throws()`, `t.doesNotThrow()` - use [**tryCatch**](https://github.com/coderaiser/try-catch) or [**tryToCatch**](https://github.com/coderaiser/try-to-catch) with [`t.equal()`](#tequalresult-any-expected-any-message-string) instead
- [`t.plan()`](https://github.com/substack/tape#tplann)

For a list of all built-in assertions, see [Operators](#operators).

## Installation

> ```plain

Reviewer: ```suggestion
```
```

   --no-check-duplicates       do not check messages for duplicates
```

> #### `--version` (alias: `-v`)

Reviewer: How this part is better then `—help`? Maybe let’s keep `—-help`, so there was one point of truth?

test('hello: world', (t) => {
    t.end();
    t.end();
    t.pass('hello, world!');
});
```

#### ✅ Example of Correct Code:

```js
test('hello: world', (t) => {
    t.pass('hello, world!');
    t.end();
});
```

### Duplicate Test Messages

📼 **Supertape** expects all test messages to be unique. This check can be disabled with:

- the [`--no-check-duplicates`](#--no-check-duplicates) command line flag
- the [`SUPERTAPE_CHECK_DUPLICATES`](#supertape_check_duplicates) environment variable (set to `0`)
- the [`checkDuplicates`](#checkduplicates-boolean) test option (set to `true`)

#### ❌ Example of Incorrect Code:

```js
test('hello: world', (t) => {
    t.pass('hello one!');
    t.end();
});

test('hello: world', (t) => {
    t.pass('hello two!');
    t.end();
});
```

### Assertions Per Test

📼 **Supertape** expects that each test contains exactly one assertion. This check can be disabled with:

- the [`--no-check-assertions-count`](#--no-check-assertions-count) command line flag
- the [`SUPERTAPE_CHECK_ASSERTIONS_COUNT`](#supertape_check_assertions_count) environment variable (set to `0`)
- the [`checkAssertionsCount`](#checkassertionscount-boolean) test option (set to `true`)

> #### ❌ Example of Incorrect Code:

Reviewer: Could you please lowercase “(in)correct code”. All examples in 🐊[**Putout**](https://github.com/coderaiser/putout) use this format, and would be great to use one format, to simplify migrations.
